homeassistant/components/template/sensor.py

```diff
@@ -237,12 +240,25 @@ def _update_state(self, result):
         )
 
 
-class TriggerSensorEntity(TriggerEntity, SensorEntity):
+class TriggerSensorEntity(TriggerEntity, RestoreSensor):
     """Sensor entity based on trigger data."""
 
     domain = SENSOR_DOMAIN
     extra_template_keys = (CONF_STATE,)
 
+    async def async_added_to_hass(self) -> None:
+        """Restore last state."""
+        await super().async_added_to_hass()
+        if (
+            (last_state := await self.async_get_last_state()) is not None
+            and (extra_data := await self.async_get_last_sensor_data()) is not None
+            and last_state.state not in (STATE_UNKNOWN, STATE_UNAVAILABLE)
+            # The trigger might have fired already while we waited for stored data,
+            # then we should not restore state
+            and CONF_STATE not in self._rendered
+        ):
+            self._rendered[CONF_STATE] = extra_data.native_value
```

Comment:
What about attributes?

Reply:
That should be implemented too, it depended on #69350 though